packages/core/src/prompts/prompts-manager.ts

```diff
@@ -0,0 +1,443 @@
+     *  - exact prompt key
+     *  - originalName (e.g., internal prompt filename without extension)
+     *  - command from front-matter (with or without leading slash)
+     */
+    async resolvePromptKey(nameOrAlias: string): Promise<string | null> {
+        if (!this.cacheValid) {
+            await this.buildPromptsCache();
+        }
+
+        // Exact key
+        if (Object.prototype.hasOwnProperty.call(this.promptsCache, nameOrAlias)) {
+            return nameOrAlias;
+        }
+
+        const want = nameOrAlias.startsWith('/') ? nameOrAlias : `/${nameOrAlias}`;
+        const wantNoSlash = nameOrAlias.startsWith('/') ? nameOrAlias.slice(1) : nameOrAlias;
+
+        for (const [key, prompt] of Object.entries(this.promptsCache)) {
+            const meta = prompt.metadata as Record<string, unknown> | undefined;
+            const originalName =
+                typeof meta?.originalName === 'string' ? (meta!.originalName as string) : undefined;
+            const command =
+                typeof meta?.command === 'string' ? (meta!.command as string) : undefined;
+
+            if (originalName === nameOrAlias) return key;
+            if (command && (command === want || command === wantNoSlash)) return key;
+        }
+
+        return null;
+    }
+
+    /**
+     * Validate prompt arguments against the prompt definition
+     */
+    private validatePromptArguments(
+        expectedArgs: PromptArgument[],
+        providedArgs: Record<string, unknown>
+    ): void {
+        const missingRequired = expectedArgs
+            .filter((arg) => arg.required === true)
+            .filter((arg) => !Object.prototype.hasOwnProperty.call(providedArgs, arg.name));
+
+        if (missingRequired.length > 0) {
+            const missingNames = missingRequired.map((arg) => arg.name);
+            throw PromptError.missingRequiredArguments(missingNames);
+        }
+
+        // Check for unknown arguments
+        const providedKeys = Object.keys(providedArgs).filter((k) => !k.startsWith('_'));
+        const expectedKeys = expectedArgs.map((arg) => arg.name);
+        const unknownArgs = providedKeys.filter((key) => !expectedKeys.includes(key));
+
+        if (unknownArgs.length > 0) {
+            logger.warn(`Unknown arguments provided: ${unknownArgs.join(', ')}`, null, 'yellow');
+        }
+    }
+
+    /**
+     * Refresh all prompt caches
+     */
+    async refresh(): Promise<void> {
+        this.invalidateCache();
+        await this.buildPromptsCache();
+        logger.info('PromptsManager refreshed');
+    }
+
+    /**
+     * Get a specific provider by source name
+     */
+    getProvider(source: string): PromptProvider | undefined {
+        return this.providers.get(source);
+    }
+
+    /**
+     * Get all registered provider sources
+     */
+    getProviderSources(): string[] {
+        return Array.from(this.providers.keys());
+    }
+
+    /**
+     * Sanitize prompt metadata to remove sensitive information
+     */
```

Comment:
i don't get this function, what is this for?

prompt.metadata is a Record<string, unknown> type

these things we are filtering out from metadata:

```
            content: _content,
            prompt: _promptText,
            filePath: _filePath,
            messages: _messages,
```

is this something specific to MCP prompts? where are these fields coming from?